BridgeBot/bidding.py

```diff
@@ -1,47 +1,53 @@
-from BridgeBot.enums import players, strains, contracts
+from enums import Players, Strains, AuctionStatus, Doubles, Contracts, PASS, Team
 
-doubles = [None, 'X', 'XX']
 
-class Auction:
-    contract = {'level': 0,
-                'strain': None,
-                'doubled': None,
-                'declarer': None}
+class InvalidDealerException(Exception):
+    pass
+
 
+class Contract:
+    def __init__(self):
+        self.level = 0
+        self.strain = Strains.NT
+        self.doubled = Doubles.NONE
+        self.declarer = None
+
+class Auction:
     def __init__(self, dealer):
         self.dealer = dealer.upper()
 
-        if not dealer in players:
-            raise Exception('Invalid dealer.')
+        if not isinstance(dealer, Players):
+            raise InvalidDealerException('Invalid dealer')
 
         self.record = {
-            'NORTH': [],
-            'EAST': [],
-            'SOUTH': [],
-            'WEST': []
+            Players.NORTH: [],
+            Players.EAST: [],
+            Players.SOUTH: [],
+            Players.WEST: []
         }
 
         self.ns_first_bid = {
```

Comment:
See the above comment about breaking out `Record` as a class. This would probably live in that class, since we want it to be updated atomically with the bidding record.

Reply:
Probably a good idea. We should try to remove the entire state of the game to its own data structure over time I think.